lightning/src/ln/chan_utils.rs

```diff
@@ -374,7 +374,7 @@ impl_writeable!(HTLCOutputInCommitment, 1 + 8 + 4 + 32 + 5, {
 });
 
 #[inline]
-pub(crate) fn get_htlc_redeemscript_with_explicit_keys(htlc: &HTLCOutputInCommitment, a_htlc_key: &PublicKey, b_htlc_key: &PublicKey, revocation_key: &PublicKey) -> Script {
+pub(crate) fn get_htlc_redeemscript_with_explicit_keys(htlc: &HTLCOutputInCommitment, local_htlc_key: &PublicKey, remote_htlc_key: &PublicKey, revocation_key: &PublicKey) -> Script {
```

Comment:
This isnt always the remote/local key though, no? We also call this with the keys swapped to get redeemscripts for *their* transactions. We probably need to standardize on some kind of nomenclature cause remote/local is often confusing.

Reply:
From the script layout as described in the BOLT3, how `local_htlcpubkey` is instanced depends if we are building a transaction for _us _ or for our counterparty.

I think when we _generate_ a transaction, including its scripts, there is no such thing as`local` or `remote` it's always a local commitment transaction. Now when we _sign_ such transaction and what we sign depends if its a local transaction or remote one.

So, as a nomenclature,  I propose:
* for all our helpers in `chan_utils` bind to local/remote to match BOLT3
* in `channel/channelmonitor` label transaction as local_tx or remote_tx (no changes expected)
* for holding our or counterparty elements such as `their_keys` in `Channel` bind to owner/counterparty avoid confusion when we instance our helpers.

We may have to move helpers from `Channel` to `chan_utils` to make it cleaner.

Reply:
The naming is definitely contextual. So I would agree that we should not settle on always one way of describing the relationship.

I would like to see "counterparty" used when generally referring to the other side (i.e., when the relationship can be flipped). I don't have a strong opinion on "owner" being the opposite of "counterparty" just yet. It may make sense to not have any prefix in that case. I'd need to see a concrete example.

I'm generally not in favor of using possessive pronouns like "our" and "their" in code. I'd prefer a more descriptive name for the relationship being modeled.

Reply:
@jkczyz fyi, even the BOLT isn't clear on this https://github.com/lightningnetwork/lightning-rfc/pull/638

> The naming is definitely contextual. So I would agree that we should not settle on always one way of describing the relationship.

I agree, do you see another context that the ones I'm describing, namely `chan_utils`, `Channel`, `channel/channelmonitor` or if those contexts make sense ? Maybe 2) and 3) are the same.

Reply:
My thinking was that if `chan_utils` is suppose to be agnostic to what is local and what is remote (at least in some places), then the "local" and "remote" naming may not make sense in those particular contexts.

So for instance, AFAICT, `TxCreationKeys` should not use the words "local" and "remote" since it is used in both contexts:

https://github.com/rust-bitcoin/rust-lightning/blob/c09c9fb9a00ebaf60cd44755e7a14df1dbb7c7a7/lightning/src/ln/channel.rs#L1102

https://github.com/rust-bitcoin/rust-lightning/blob/c09c9fb9a00ebaf60cd44755e7a14df1dbb7c7a7/lightning/src/ln/channel.rs#L1116

Additionally, there is a lot of redundancy in some of the naming that isn't necessary in the given context. e.g., `initial_local_commitment_tx.local_keys.local_htlc_key`.

Reply:
Right there's kind of two different things we'll want two words for:
 * Something is our's/our counterparty's (I think I like ours/counterpartys or so?)
 * Something is for the "owner of the transaction", which is what local/remote mean here.

Its true the BOLTs are a mess in this regard, but thats precisely why we should try to do better.

As a concrete proposal, I think ours/counterpartys and local/remote (which is whats there already in TxCreationKeys) is reasonable, but am open to other ideas.

Reply:
For `ours` and `counterpartys`, I would prefer no prefix and `counterparty`.